src/atlas/grid/detail/partitioner/CubedSpherePartitioner.cc

```diff
@@ -0,0 +1,248 @@
+                cb.globalProcEndPE[t] =  cb.globalProcEndPE[t-1];
+            } else {
+                cb.globalProcStartPE[t] = cb.globalProcEndPE[t-1] + 1;
+                cb.globalProcEndPE[t] =   cb.globalProcStartPE[t] + cb.nproc[t] - 1;
+            }
+        }
+
+
+    }  else {
+        for (size_t t = 0; t < 6; ++t) {
+            cb.globalProcStartPE[t] = globalProcStartPE_[t];
+            cb.globalProcEndPE[t] = globalProcEndPE_[t];
+            cb.nprocx[t] = nprocx_[t];
+            cb.nprocy[t] = nprocy_[t];
+        }
+        cb.nproc[0] = globalProcEndPE_[0] + 1;
+        for (size_t t = 1; t < 6; ++t) {
+            cb.nproc[t] = cb.globalProcStartPE[t] == cb.globalProcEndPE[t-1] ? cb.nproc[t] = 0 :
+                    cb.globalProcEndPE[t] - cb.globalProcStartPE[t] + 1;
+        }
+
+    }
+    return cb;
+}
+
+void CubedSpherePartitioner::partition( CubedSphere& cb, const int nb_nodes, const CellInt nodes[], int part[] ) const {
+
+    // ngpt number of grid points per tile (this is for cell centers where the number of cells are the same
+    std::size_t tileCells = static_cast<std::size_t>(nb_nodes/6);
+    std::array<std::size_t, 6> ngpt{tileCells, tileCells, tileCells, tileCells, tileCells, tileCells};
+
+    // xoffset - the xoffset per band per tile (including cb.nx[t])
+    //std::vector<std::vector<atlas::idx_t>> xoffset(createOffset(ngpt, cb.nprocx, cb.nx, cb.ny));
+
+    cb.xoffset = createOffset(ngpt, cb.nprocx, cb.nx, cb.ny);
+
+    // yoffset - the yoffset per band per tile (including cb.ny[t])
+    cb.yoffset = createOffset(ngpt, cb.nprocy, cb.ny, cb.nx);
+
+    int p;  // partition index
+    // loop over all data tile by tile
+    for (int n = 0; n < nb_nodes; ++n) {
+        std::size_t t = static_cast<std::size_t>(nodes[n].t);
+        p = cb.globalProcStartPE[t];
+        for (std::size_t yproc = 0; yproc < static_cast<std::size_t>(cb.nprocy[t]); ++yproc) {
+            for (std::size_t xproc = 0; xproc < static_cast<std::size_t>(cb.nprocx[t]); ++xproc) {
+                if ( (nodes[n].y >= cb.yoffset[t][yproc]) && (nodes[n].y < cb.yoffset[t][yproc+1]) &&
+                     (nodes[n].x >= cb.xoffset[t][xproc]) && (nodes[n].x < cb.xoffset[t][xproc+1]) )
+                {
+                    part[n] = p;
+                }
+                ++p;
+            }
+        }
+    }
+    ATLAS_ASSERT( part[nb_nodes-1] == nb_partitions() - 1, "number of partitions created not what is expected");
+}
+
+void CubedSpherePartitioner::partition( const Grid& grid, int part[] ) const {
+    if ( nb_partitions() == 1 )  // trivial solution, so much faster
+    {
+        for ( idx_t j = 0; j < grid.size(); ++j ) {
+            part[j] = 0;
+        }
+    }
+    else {
+        auto cb = cubedsphere( grid );
+
+        std::vector<CellInt> nodes( static_cast<std::size_t>(grid.size()) );
+        std::size_t n( 0 );
+
+        for ( std::size_t it = 0; it < 6; ++it ) {
```

Comment:
This is really a really nitpicky formatting point, but, as I think has been noted elsewhere, the use of spaces near parentheses seems inconsistent. For example, the `for` loop here has the spaces, whereas the `for` loops elsewhere (_e.g._ lines 198-199) do not. Is there a coding standards document that I should be aware of when I write my code?